### PR TITLE
Redesign account menu

### LINKED ISSUE #6101: Redesign Account Menu
# Description

## User Stories

* > As a user I want to find all account related settings at one spot so that I don't need to search for them.

## Value
- self service

## Acceptance Criteria
- click on Avatar / Initials opens drop menu
Options are:
1. **Profile:** Avatar / Initials, display name (Fallback: username), email adress (nav item to profile page)
2. **Settings** (nav item to settings page)
3. **Quota:** Not clickable for now, no hover. Shows relative (integer % - no decimals) and absolute quota (human readable filesize/quota, see filesize component, make sure locale is taken into account). Progress bar: changes color according to progress. 0-79%: owncloud blue-gradient; 80-89%: warning-color (yellow) 90-100%: Error-color (red). 
4. **Logout**

Side note: via graph API we'll have the quota warning stages via API

## Definition of done
- Functional requirements
[ ] functionality described in the user story works
[ ] acceptance criteria are fulfilled
- Quality
[ ] codre review happened
[ ] CI is green
[ ] critical code received unit tests by the developer
[ ] automated tests passed (if automated tests are not available, this test needs to be created and passed
- Non-functional requirements
[ ] no sonar cloud issues

### PR DESCRIPTION
**Pull request by lookacat**
## Description
See #6101 

## Screenshots
Screenshot 2022-01-28 at 08 45 07
Screenshot 2022-01-28 at 08 45 35

## Related Issue

- Fixes #6101 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: packages/web-runtime/src/components/UserMenu.vue [hunk 1]
```diff
@@ -26,64 +25,118 @@
...
+          </li>
+        </oc-list>
+      </template>
     </oc-drop>
   </nav>
 </template>
 
 <script>
 import NavigationMixin from '../mixins/navigationMixin'
+import { mapGetters } from 'vuex'
```
Comment (pascalwengerter) [inline packages/web-runtime/src/components/UserMenu.vue:139]: Why is this necessary?

File: packages/web-runtime/src/components/UserMenu.vue [hunk 2]
```diff
@@ -96,13 +149,49 @@ export default {
...
   },
   computed: {
+    ...mapGetters(['quota']),
+
+    quotaTotal() {
+      // Example return 500 * 1024 * 1024 * 1024
+      return this.quota.total
+    },
+    quotaUsed() {
+      // Example return 500 * 0.8 * 1024 * 1024 * 1024
```
Comment (pascalwengerter) [inline packages/web-runtime/src/components/UserMenu.vue:171]: ?

File: packages/web-runtime/src/components/UserMenu.vue [hunk 3]
```diff
@@ -96,13 +149,49 @@ export default {
...
       type: Array,
       required: false,
       default: () => []
     }
   },
   computed: {
+    ...mapGetters(['quota']),
+
+    quotaTotal() {
+      // Example return 500 * 1024 * 1024 * 1024
```
Comment (pascalwengerter) [inline packages/web-runtime/src/components/UserMenu.vue:167]: ?

File: packages/web-runtime/src/components/UserMenu.vue [hunk 4]
```diff
@@ -96,13 +149,49 @@ export default {
...
+      return this.byteToGb(this.quotaTotal)
+    },
+    quotaUsageGb() {
+      return this.byteToGb(this.quotaUsed)
+    },
+    quotaUsagePercent() {
+      return (this.quotaUsed / this.quotaTotal) * 100
+    },
+    quotaProgressVariant() {
+      switch (true) {
```
Comment (pascalwengerter) [inline packages/web-runtime/src/components/UserMenu.vue:184]: ?

File: packages/web-runtime/src/components/UserMenu.vue [hunk 5]
```diff
@@ -26,64 +25,118 @@
...
+      </template>
     </oc-drop>
   </nav>
 </template>
 
 <script>
 import NavigationMixin from '../mixins/navigationMixin'
+import { mapGetters } from 'vuex'
 
 export default {
```
Comment (pascalwengerter) [inline packages/web-runtime/src/components/UserMenu.vue:141]: The component doesn't seem to have tests, would you mind adding some?

File: packages/web-runtime/src/components/UserMenu.vue [hunk 6]
```diff
@@ -114,6 +203,12 @@ export default {
     })
   },
   methods: {
+    byteToGb(value) {
```
Comment (pascalwengerter) [inline packages/web-runtime/src/components/UserMenu.vue:206]: Pretty sure we handle this with an external library or a helper somewhere, let me check

**Review by pascalwengerter — CHANGES_REQUESTED** [review]
Needs a rebase and perhaps adjustments for the (by now) merged theme switcher, but looking pretty awesome already!

File: packages/web-runtime/src/components/UserMenu.vue (hunk 6 above)
Comment (kulmann) [reply packages/web-runtime/src/components/UserMenu.vue:206]: `filesize`

File: packages/web-runtime/src/components/UserMenu.vue [hunk 7]
```diff
@@ -26,64 +25,118 @@
...
-            justify-content="left"
-            @click="logout"
-          >
-            <oc-icon name="logout-box-r" fill-type="line" />
-            <translate>Log out</translate>
-          </oc-button>
-        </li>
-      </ul>
+      <template #special>
+        <oc-list raw="true" class="oc-pb-l oc-pt-s oc-pl-s oc-pr-s">
```
Comment (pascalwengerter) [inline packages/web-runtime/src/components/UserMenu.vue:29]:
```suggestion
        <oc-list :raw="true" class="oc-pb-l oc-pt-s oc-pl-s oc-pr-s">
```

Otherwise Vue complains about receiving a string and expecting a bool ;)

File: packages/web-runtime/src/components/UserMenu.vue (hunk 1 above)
Comment (lookacat) [reply packages/web-runtime/src/components/UserMenu.vue:139]: because of the quota

File: packages/web-runtime/src/components/UserMenu.vue (hunk 2 above)
Comment (lookacat) [reply packages/web-runtime/src/components/UserMenu.vue:171]: was just to test around, dev left over

File: packages/web-runtime/src/components/UserMenu.vue [hunk 8]
```diff
@@ -123,3 +218,79 @@ export default {
   }
 }
 </script>
+
+<style lang="scss" scoped>
+/* TODO: Replace hard coded colors */
```
[the remark this answers is not in the document]
Comment (lookacat) [reply packages/web-runtime/src/components/UserMenu.vue:223]: now :D

File: packages/web-runtime/src/components/UserMenu.vue (hunk 6 above)
Comment (lookacat) [reply packages/web-runtime/src/components/UserMenu.vue:204]: ok

File: packages/web-runtime/src/components/UserMenu.vue (hunk 7 above)
Comment (lookacat) [reply packages/web-runtime/src/components/UserMenu.vue:29]: okok i see :D

File: packages/web-runtime/src/components/UserMenu.vue [hunk 9]
```diff
@@ -26,64 +25,118 @@
...
+              variation="inverse"
+              gap-size="xsmall"
+              justify-content="left"
+              class="row"
+              @click="logout"
+            >
+              <div class="icon-wrapper oc-pr-s">
+                <oc-icon name="logout-box-r" fill-type="line" />
+              </div>
+              <translate>Log out</translate>
```
[the remark this answers is not in the document]
Comment (lookacat) [reply packages/web-runtime/src/components/UserMenu.vue:101]: tobi changed it to be the last ^^

File: packages/web-runtime/src/components/UserMenu.vue (hunk 6 above)
Comment (kulmann) [reply packages/web-runtime/src/components/UserMenu.vue:200]: for what it's worth, there is a `OcResourceSize` component in ODS which handles all the conversion (uses `filesize` internally as well) and creates a span if I remember correctly.

**Comment by lookacat** [comment]
rebase done, most stuff fixed now, will now adjust the color vars and add/fix tests

File: packages/web-runtime/src/components/UserMenu.vue (hunk 1 above)
Comment (pascalwengerter) [reply packages/web-runtime/src/components/UserMenu.vue:81]: Yeah sry, I'm an idiot at times

File: packages/web-runtime/src/components/UserMenu.vue (hunk 5 above)
Comment (pascalwengerter) [reply packages/web-runtime/src/components/UserMenu.vue:84]: Will add them myself tomorrow

File: packages/web-runtime/src/components/Topbar/UserMenu.vue [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+          >
+            <avatar-image :width="32" :userid="userId" :user-name="user.displayname" />
+            <span class="profile-info-wrapper" :class="{ 'oc-py-xs': !user.email }">
+              <span v-text="user.displayname" />
+              <br v-if="user.email" />
+              <span v-if="user.email" class="oc-text-small" v-text="user.email" />
+            </span>
+          </oc-button>
+        </li>
+        <li v-for="(n, nid) in applicationsList" :key="`user-menu-${nid}`">
```
Comment (fschade) [inline packages/web-runtime/src/components/Topbar/UserMenu.vue:42]: not a fan of using shorthands like `n` and `nid`. I would prefer to have something human readable here

File: packages/web-runtime/src/components/Topbar/UserMenu.vue [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+      return filesize(this.quota.total)
+    },
+    quotaUsed() {
+      return filesize(this.quota.used)
+    },
+    quotaUsagePercent() {
+      return ((this.quota.used / this.quota.total) * 100).toFixed(2)
+    },
+    quotaProgressVariant() {
+      if (this.quotaUsagePercent < 80) return 'primary'
```
Comment (fschade) [inline packages/web-runtime/src/components/Topbar/UserMenu.vue:121]: would it make sense to have the thresholds as an settings value? I am not sure if they are always the same

File: packages/web-runtime/src/store/user.js [hunk 12]
```diff
@@ -233,10 +226,16 @@ const actions = {
 
 const mutations = {
   SET_USER(state, user) {
+    let email
+    if (Object.keys(user.email).length === 0) {
```
Comment (fschade) [inline packages/web-runtime/src/store/user.js:230]: I wonder why actual email can be an object...

**Review by fschade — APPROVED** [review]
I only found a few small things, otherwise great

File: packages/web-runtime/src/components/Topbar/UserMenu.vue (hunk 11 above)
Comment (pascalwengerter) [reply packages/web-runtime/src/components/Topbar/UserMenu.vue:121]: Product question for @tbsbdr, sounds like a lot of complexity for something no one asked for so far:)

File: packages/web-runtime/src/components/Topbar/UserMenu.vue (hunk 10 above)
Comment (pascalwengerter) [reply packages/web-runtime/src/components/Topbar/UserMenu.vue]: Changed to `app, index`

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingAccept</strong> https://drone.owncloud.com/owncloud/web/22248/15/1
The following scenarios passed on retry:
- webUISharingAcceptSharesToRoot/acceptShares.feature:108

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingInternal3</strong> https://drone.owncloud.com/owncloud/web/22248/59/1
The following scenarios passed on retry:
- webUISharingInternalUsersSharingIndicator/shareWithUsers.feature:17

**Review by pascalwengerter — APPROVED** [review]
